hax/hax/ha/events.py

```diff
@@ -104,7 +107,14 @@ def get_next_message(self, time_out) -> Optional[Event]:
         acknowledge that message is already processed
         """
         logging.debug('Listening......')
-        message = self.consumer.receive(time_out)
+        try:
+            # HA container may come up later than hax container, and then
+            # register the topic online. We need to wait for it to be
+            # available.
+            message = self.consumer.receive(time_out)
+        except Exception:
+            logging.warning('Subscribed topic not available. Waiting...')
```

Comment:
Sorry I didn't get what do we mean by `Waiting...` here, do we block somehow?

Reply:
Waiting as in- `ha-event-listener` thread will be running in a similar fashion like when we are waiting for any new message on the message bus.

```
while not self.stopped:
                self._handle_next_messages()
                wait_for_event(self.event, self.interval_sec)
```

Reply:
Reason we are doing this is because

> If HA hasn't registered the topic name - as consumers we will face this issue
2021-12-09 16:35:08,416 [ERROR] {ha-event-listener} [receive] MessageBusError: 4352 poll(0.1) for consumer hare failed to receive message. KafkaError{code=UNKNOWN_TOPIC_OR_PART,val=3,str="Subscribed topic not available: ha_event_hare: Broker: Unknown topic or partition"}
2021-12-09 16:35:08,416 [ERROR] {ha-event-listener} Aborting due to an error

Had a discussion with Shailesh Chaudhari, about handling the above scenario.

- There will be no ordering constraints, nodes (pods) could come up in any order.
- When HA containers are online and have registered the topic online. Only then hare's consumer thread can start listening to that topic, else we see the above error.
- Earlier, hare's consumer thread aborted in case of failure of availability of topic. 
- Now, Hare needs to wait for the topic to be available.
- Once the subscribe HA API is available, the subscription API transaction can make sure that the topic exists before HA subscribe API returns it. But still, even there, HA needs to come up and REST API to be available, it may come up later than storage or server pod where hax is running (unlikely but possible).